### PR TITLE
chore: refactor + update to v4

### PR DESCRIPTION
**Pull request by benouat**
Missing part would be adding:

- https://github.com/marionebl/commitlint

### CODE REVIEW

**Comment by benouat** [comment]
cc @pkozlowski-opensource @maxokorokov if you wan't to have a look... feel free to ask for more details if needed.

Next step would be to publish`v4.0.0` as soon as merged.

File: src/utils/testing/test-app.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export function createTestApp(
+  runner: SchematicTestRunner,
+  appOptions = {}
+): UnitTestTree {
+  const workspaceTree = runner.runExternalSchematic(
+    "@schematics/angular",
+    "workspace",
+    {
+      name: "workspace",
+      version: "6.0.0",
```
Comment (maxokorokov) [inline src/utils/testing/test-app.ts:13]: What is this version, shouldn't it be '7.0.0' ?

File: package.json [hunk 2]
```diff
@@ -11,16 +11,26 @@
...
-    "clean": "rm -rf dist",
-    "build": "rm -rf dist && tsc && node ./create_package.js && cp src/collection.json dist"
+    "test": "jest",
+    "clean": "rm -Rf dist",
+    "build": "yarn clean && mkdir -p dist && node scripts/create_package.js && tsc -p tsconfig.json"
+  },
+  "dependencies": {
+    "@angular-devkit/core": "^7.0.6",
+    "@angular-devkit/schematics": "^7.0.6",
+    "@schematics/angular": "^7.0.6",
```
Comment (maxokorokov) [inline package.json:27]: I wonder if there any relation between Angular version, CLI version and our dependencies in `dist/package.json`:

```json
"dependencies": {
    "@angular-devkit/core": "^7.0.6",
    "@angular-devkit/schematics": "^7.0.6",
    "@schematics/angular": "^7.0.6",
    "typescript": "~3.1.4"
  }
```

**Review by maxokorokov — COMMENTED** [review]
LGTM overall!

I tired:
- following the readme to run tests → OK
- running schematics on a new project with 7.1 → OK

I didn't go deep into implementation details, but now we can use tools from `@schematics/angular/utility`, which is nice!

Comments:
- CLI has the following options when creating a new project:
screen shot 2018-12-13 at 11 39 27

I guess now we support only CSS and SCSS, should we add others ? 
It doesn't crash with SASS, for instance, just doesn't add any styles to a CLI project.

**Pull request merged**